axelrod/graph.py

```diff
@@ -0,0 +1,126 @@
+"""
+Weighted undirected sparse graphs.
+
+Original source:
+https://github.com/marcharper/stationary/blob/master/stationary/utils/graph.py
```

Comment:
To be clear: If we go with my suggestion of `3.i` then that would be work for a different PR.